src/fft_simd.rs

```diff
@@ -214,7 +214,18 @@ impl FftSimd<c64> for Scalar {
 
     #[inline(always)]
     fn mul(self, a: c64, b: c64) -> c64 {
-        a * b
+        let ab = a;
+        let xy = b;
+
+        let a = ab.re;
+        let b = ab.im;
+        let x = xy.re;
+        let y = xy.im;
+
+        c64 {
+            re: f64::mul_add(a, x, -b * y),
+            im: f64::mul_add(a, y, b * x),
+        }
```

Comment:
what about 

```rust
            re: f64::mul_add(a.re, b.re, -a.im * b.im),
            im: f64::mul_add(a.re, b.im, a.im * b.re),
```

Reply:
yeah, i could do that

Reply:
not strictly required, was just a bit surprised by the way it was written